pkg/acsengine/transform.go

```diff
@@ -192,3 +195,63 @@ func removeImageReference(logger *logrus.Entry, resourceProperties map[string]in
 	}
 	return ok
 }
+
+// NormalizeResourcesForK8sMasterUpgrade takes a template and removes elements that are unwanted in any scale up/down case
+func NormalizeResourcesForK8sMasterUpgrade(logger *logrus.Entry, templateMap map[string]interface{}) error {
+	var computeResourceTypes = []string{vmResourceType, vmExtensionType}
+
+	for _, computeResourceType := range computeResourceTypes {
+		resources := templateMap[resourcesFieldName].([]interface{})
+
+		agentPoolIndex := -1
+		//remove agent nodes resources
+		for index, resource := range resources {
+			resourceMap, ok := resource.(map[string]interface{})
+			if !ok {
+				logger.Warnf("Template improperly formatted")
+				continue
+			}
+
+			resourceType, ok := resourceMap[typeFieldName].(string)
+			if !ok || resourceType != computeResourceType {
+				continue
+			}
+
+			resourceName, ok := resourceMap[nameFieldName].(string)
+			if !ok {
+				logger.Warnf("Template improperly formatted")
+				continue
+			}
+
+			if strings.EqualFold(resourceType, vmResourceType) &&
+				strings.Contains(resourceName, "variables('masterVMNamePrefix')") {
+				resourceProperties, ok := resourceMap[propertiesFieldName].(map[string]interface{})
+				if !ok {
+					logger.Warnf("Template improperly formatted")
```

Comment:
I was following the convention of other methods in this file of continuing. I haven't actively given it any thought (yet). I am going to leave it as is for now. This is one of those error cases that need handling.